docs/docs/docs/01-core/configuration/00-intro.md

```diff
@@ -0,0 +1,115 @@
+---
+title: Introduction
+tags:
+  - configuration
+---
+
+# Configuration
+
+This section is an overview of how configuration files are loaded in Strapi and how they should be used.
+
+## Config loading
+
+Loading of configuration is the very first thing Strapi does in bootstrap (after setting signal handlers), which is necessary because most of the Strapi bootstrap process depends on values from the configuration.
+
+This means that configuration loading happens even before Strapi creates a logger. Therefore any errors or warnings from the configuration unfortunately must output to `console`
+
+NOTE: For now. in theory, we could define some configuration files that do not depend on being passed a `strapi` object and could provide logger settings in there, but at this time we do not have that capability.
+
+Currently, Strapi loads every .js and .json file within a project's `./config` directory directly into the strapi config object. That is, if there is a file called `./config/myconfig.js` it will be loaded automatically and accessible from `strapi.config.get('myconfig')`
+
+(COMING SOON) In v5, Strapi will load every defined value from the `STRAPI_` environment namespace into the configuration, using underscore (\_) as the object delimiter. For example, if you define `STRAPI_ADMIN_APP_KEYS=abcd,efg` it will be available from `strapi.config.get('admin.app.keys')` as `['abcd', 'efg']`
+
+Along with that feature, all configuration values will become case-insensitive. That is, accessing `strapi.config.get('admin.App.Keys')` will return the same value as `strapi.config.get('admin.app.keys')`.
+
+## Base Strapi Configurations
+
+The following are considered 'base' Strapi configurations and a description of what each file contains is given, so that new settings can be placed in the appropriate place.
+
+Starting in Strapi v5, the contents of the base configurations will be strictly defined and no values that are not created by Strapi will be allowed in them; that is, an error will be thrown on unrecognized values. Any user additions must go in separate configuration namespaces.
```

Comment:
Are we gonna keep saying Strapi v5 or shouldn't we also move to Strapi 5

Reply:
This is targeting v5 so this part could be worded differently, yeah. I'll rewrite it with that in mind.

The issue above (where I put coming soon) is that we may not have it for v5 launch, so I wanted to include the note for the future to explain the motiviations.

Reply:
Oh, do you mean the 'v5 of Strapi' and 'Strapi 5' specifically? With that I think it's fine either way in contributor docs, we have lots of "v4" as well.

Reply:
yeah it's not a big deal